src/presign/participant.rs

```diff
@@ -97,9 +96,66 @@ pub(crate) struct PresignParticipant {
     broadcast_participant: BroadcastParticipant,
 }
 
+/// Input needed for [`PresignParticipant`] to run.
+pub(crate) struct Input {
+    /// The private keyshare of this participant.
+    keyshare_private: KeySharePrivate,
+    /// The public keyshares of all the participants.
+    all_keyshare_public: Vec<KeySharePublic>,
+    /// The private auxinfo of this participant.
+    auxinfo_private: AuxInfoPrivate,
+    /// The public auxinfo of all the participants.
+    all_auxinfo_public: Vec<AuxInfoPublic>,
+}
+
+impl Input {
+    /// Creates a new [`Input`] from the outputs of
+    /// [`crate::auxinfo::participant::AuxInfoParticipant`]
+    /// and [`crate::keygen::participant::KeygenParticipant`].
+    pub(crate) fn new(
+        all_auxinfo_public: Vec<AuxInfoPublic>,
+        auxinfo_private: AuxInfoPrivate,
+        all_keyshare_public: Vec<KeySharePublic>,
+        keyshare_private: KeySharePrivate,
+    ) -> Self {
```

Comment:
suggestion: check that the two vecs are the same length? We don't have any point of truth re: the correct length here, but they should at least be the same.